applicationset/generators/cluster.go

```diff
@@ -179,7 +179,7 @@ func (g *ClusterGenerator) getSecretsByClusterName(appSetGenerator *argoappsetv1
 	res := map[string]corev1.Secret{}
 
 	for _, cluster := range clusterSecretList.Items {
-		clusterName := string(cluster.Data["name"])
+		clusterName := string(cluster.Data["server"])
 
 		res[clusterName] = cluster
```

Comment:
IMO I think the chosen approach is quite intrusive, and a simpler, less risky way would be to just check if the `clusterName` already exists in the map.

```suggestion
		if _, ok := res[clusterName]; ok {
		  return nil, fmt.Errorf("clusterName '%s' already present in secret list", clusterName)
                }
		res[clusterName] = cluster                
```